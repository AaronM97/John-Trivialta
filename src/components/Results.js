import React from 'react'

export default function Results(props) {
  return (
    <div>
      <h1>Results!!!!</h1>
      <img src="https://akns-images.eonline.com/eol_images/Entire_Site/2013625/rs_560x415-130725124224-1024..travolta.ls.72513_copy_2.jpg?fit=inside|900:auto&output-quality=90" alt="john travolta celebrating" />
      <p>Your score is: {props.score} / 10</p>
      <a href="/"><h4>Start again</h4></a>
    </div>
  )
}
