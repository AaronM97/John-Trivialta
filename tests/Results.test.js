import React from 'react';
import ReactDOM from 'react-dom';
import Results from '../src/components/Results';
import Enzyme, { shallow } from 'enzyme';

describe('renders Results.js without crashing', () => {
    // needs a score to render. makes dummy data to check if the score is there in order for the results with a div inside to render
    it('smoke test', () => {
        let simData = {score: 1};
        const div = document.createElement('div');
        ReactDOM.render(<Results score={simData.score} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    // sets a dummy data score. checks if we can find within a p tag the dummy data score
    it('renders the CORRECT score', () => {
        let simData = {score: 827234};
        const wrapper = shallow (<Results score={simData.score} />);
        expect(wrapper.find('p').text()).toEqual('Your score is: 827234 / 10');
    })
})