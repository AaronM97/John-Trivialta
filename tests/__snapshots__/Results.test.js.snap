// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Results.js without crashing should render correctly 1`] = `
<div>
  <h1>
    Results!!!!
  </h1>
  <p>
    Your score is: 
     / 10
  </p>
  <a
    href="/"
  >
    <h4>
      Start again
    </h4>
  </a>
</div>
`;
